cmds/subvolume.c

```diff
@@ -347,6 +347,7 @@ static const char * const cmd_subvolume_delete_usage[] = {
 	OPTLINE("-c|--commit-after", "wait for transaction commit at the end of the operation"),
 	OPTLINE("-C|--commit-each", "wait for transaction commit after deleting each subvolume"),
 	OPTLINE("-i|--subvolid", "subvolume id of the to be removed subvolume"),
+	OPTLINE("-R|--recursive", "delete subvolumes beneath each subvolume recursively"),
```

Comment:
This must mention the conditions or limitations when it will not remove the subvolumes.

Reply:
I do not think it's a good idea to put all details into the normally-one-line help line.

The details including the privilege/mount option requirement should go the man page.
Or every one-line help string will be no different than the man page.

Reply:
Missing key information about the behaviour of the option is serious problem in usability, nothing even hints that there are exceptions so no reason to look to the manual page. The text does not need to be on one line, the OPTLINE is there to format the text and there are options with long text, eg 'check --force'.

Reply:
Your argument doesn't stand, especially there is *no* mention in the `btrfs subvolume delete` help string at all, on the privilege nor the mount option.

In fact the privilege/mount option requirement are all the same as regular subvolume deletion.
You're asking for details that should be included as generic requirement to be added to a specific option.

And even your privilege/mount option argument makes no sense for the recursive option.
Just go check the help string from `rm`. Does it even mention anything like privilege?

Let me be clear, if you want to add mount option/privilege for subvolume deletion, go add it to the generic help string of 'btrfs subvolume delete', not to this specific option.

If you want the recursive specific implementation details, I'm fine with the non-atomic part into the man page.

If you want the privilege/mount option/non-atomic details to be all added into help string, I'll call it BS.